---
title: C28105
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C28105
helpviewer_keywords:
- C28105
ms.assetid: 5afdec78-30eb-470c-980f-93d0cb3e74cc
author: corob-msft
ms.author: corob
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: 35f138288cc46e1b7a4be34b530f07df7f66caff
ms.sourcegitcommit: 68f893f6e472df46f323db34a13a7034dccad25a
ms.translationtype: MT
ms.contentlocale: zh-TW
ms.lasthandoff: 02/15/2020
ms.locfileid: "77270186"
---
# <a name="c28105"></a>C28105
警告 C28105：因例外狀況而洩漏資源

 當引發例外狀況時，不會釋放指定的資源。 路徑所指定的語句可能會引發例外狀況。 此警告類似于警告[C28103](../code-quality/c28103.md)，但在此情況下，會牽涉到例外狀況。

## <a name="example"></a>範例
 下列程式碼範例會產生此警告：

```cpp
res = KeSaveFloatingPointState(buffer);

res = AllocateResource(Resource);
char *p2 = new char[10]; // could throw

delete[] p2;
FreeResource(Resource)
```

 下列程式碼範例可避免此警告：

```cpp
res = AllocateResource(Resource);
char *p2;

try {
    p2 = new char[10];
} catch (std::bad_alloc *e) {
    // just handle the throw
    ;
}
FreeResource(Resource)
```
